### ISSUE TITLE
[rllib] SAC does not multiply importance weights when calculating loss

### ISSUE THREAD

**Issue by Purpleslz** [ray-project/ray#11591]
SAC should multiply importance weights here:
```
    critic_loss = [
        0.5 * torch.mean(torch.pow(q_t_selected_target - q_t_selected, 2.0)) 
        # <- here should multiply importance weights like ddpg/dqn implementation.
    ]
```
SAC code(without multiplying importance weights): https://github.com/ray-project/ray/blob/master/rllib/agents/sac/sac_torch_policy.py#L265

DQN/DDPG code(with multiplying importance weights): https://github.com/ray-project/ray/blob/4ccfd07a614c3b0620e75ccb47c327bb0f817299/rllib/agents/dqn/dqn_torch_policy.py#L98
https://github.com/ray-project/ray/blob/master/rllib/agents/ddpg/ddpg_torch_policy.py#L140

**Comment by sven1977** [comment]
Yes, you are right. This is a bug. Will provide a fix PR.

**Comment by sven1977** [comment]
@Purpleslz 
PR: https://github.com/ray-project/ray/pull/12394

Added Huber-loss as well.
